plugins/amazonq/mynah-ui/src/mynah-ui/ui/main.ts

```diff
@@ -671,147 +740,173 @@ export const createMynahUI = (
+                    return
+                }
 
-    followUpsInteractionHandler = new FollowUpInteractionHandler({
-        mynahUI,
-        connector,
-        tabsStorage,
-    })
-    quickActionHandler = new QuickActionHandler({
-        mynahUI,
-        connector,
-        tabsStorage,
-        isFeatureDevEnabled,
-        isCodeTransformEnabled,
-        isDocEnabled,
-        isCodeScanEnabled,
-        isCodeTestEnabled,
-    })
-    textMessageHandler = new TextMessageHandler({
-        mynahUI,
-        connector,
-        tabsStorage,
-    })
-    messageController = new MessageController({
-        mynahUI,
-        connector,
-        tabsStorage,
-        isFeatureDevEnabled,
-        isCodeTransformEnabled,
-        isDocEnabled,
-        isCodeScanEnabled,
-        isCodeTestEnabled,
-    })
+                this.connector?.onCustomFormAction(tabId, messageId, action, eventId)
+            },
+            defaults: {
+                store: this.tabDataGenerator.getTabData('cwc', true),
+            },
+            config: {
+                maxTabs: 10,
+                feedbackOptions: feedbackOptions,
+                texts: uiComponentsTexts,
+            },
+        }
+        if (!hybridChat) {
+            /**
+             * when in hybrid chat the reference gets resolved later so we
+             * don't need to create mynah UI
+             */
+            this.mynahUIRef = { mynahUI: new MynahUI({ ...this.mynahUIProps, loadStyles: false }) }
+        }
+        this.followUpsInteractionHandler = new FollowUpInteractionHandler({
+            mynahUIRef: this.mynahUIRef,
+            connector: this.connector,
+            tabsStorage: this.tabsStorage,
+        })
+
+        this.textMessageHandler = new TextMessageHandler({
+            mynahUIRef: this.mynahUIRef,
+            connector: this.connector,
+            tabsStorage: this.tabsStorage,
+        })
+        this.messageController = new MessageController({
+            mynahUIRef: this.mynahUIRef,
+            connector: this.connector,
+            tabsStorage: this.tabsStorage,
+            isFeatureDevEnabled,
+            isCodeTransformEnabled,
+            isDocEnabled,
+            isCodeScanEnabled,
+            isCodeTestEnabled,
+        })
+        this.quickActionHandler = new QuickActionHandler({
+            mynahUIRef: this.mynahUIRef,
+            connector: this.connector!,
+            tabsStorage: this.tabsStorage,
+            isFeatureDevEnabled: this.isFeatureDevEnabled,
+            isCodeTransformEnabled: this.isCodeTransformEnabled,
+            isDocEnabled: this.isDocEnabled,
+            isCodeScanEnabled: this.isCodeScanEnabled,
+            isCodeTestEnabled: this.isCodeTestEnabled,
+            hybridChat
+        })
+
+    }
+    get mynahUI(): MynahUI | undefined {
+        return this.mynahUIRef.mynahUI
+    }
 }
+
+
+export type MynahUIRef = { mynahUI: MynahUI | undefined }
```

Comment:
why undefined?

Reply:
it is lazily injected once the connector is initialized, so it'll be undefined before